### PR TITLE
Bug Fix

### PR DESCRIPTION
**Pull request by syeda-farha-2023**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,75 +1,63 @@
...
+                    def directories = sh(script: 'find . -mindepth 1 -maxdepth 1 -type d', returnStdout: true).trim().split('\n')
+                    for (dir in directories) {
+                        dir(dir) {
+                            // Check if deploy.sh exists and execute it
+                            if (fileExists('deploy.sh')) {
+                                sh './deploy.sh'
+                            } else {
+                                // Use a generic deployment command if no deploy.sh exists
+                                echo "No deploy.sh found in ${dir}, using default gcloud deploy command"
+                                sh "gcloud functions deploy --runtime python39 --trigger-http --allow-unauthenticated"
```
Comment (sagar-vaghela97) [inline Jenkinsfile:36]: In the else, throw error and fail the pipeline. We do not have any default deploy functionality.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,75 +1,63 @@
 pipeline {
     agent any
-    
     environment {
-        GOOGLE_APPLICATION_CREDENTIALS = credentials('service-account') // Replace 'service-account' with your Jenkins credentials ID
-        PROJECT_ID = 'devops-testing-419206' // Replace with your actual project ID
-        REGION = 'asia-south1' // Replace with your actual region
+        // Deployment configuration
+        DEPLOY_TARGET = 'staging'
+        FEATURE_X_ENABLED = 'true'
```
Comment (sagar-vaghela97) [inline Jenkinsfile:6]: What is FEATURE_X_ENABLED?

File: Jenkinsfile (hunk 1 above)
Comment (syeda-farha-2023) [reply Jenkinsfile:36]: Based on the repository structure, which may contain various folders and cloud functions, I initially included an else block to use a default deployment command for testing purposes if no deploy functionality is found.

File: Jenkinsfile (hunk 2 above)
Comment (syeda-farha-2023) [reply Jenkinsfile:6]: FEATURE_X_ENABLED = 'true' is like a switch. When this switch is set to 'true', it tells the cloud functions to turn on "Feature X" during deployment. If the functions or deployment scripts check this setting. As a result, they might enable new logging system, additional features, configurations, or behaviors. If it were set to 'false', "Feature X" would stay off.

I add this functionality only for a deployment configuration.

**Pull request closed without merging**